pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -338,61 +338,68 @@ func (e *EndpointController) syncService(key string) error {
 
 	readyEps := 0
 	notReadyEps := 0
+	addEndpointSubset := func(pod *v1.Pod, epa v1.EndpointAddress, epp v1.EndpointPort) {
```

Comment:
I really do not like to have a function variable and let it manipulate local vars in line. Can we separate this into 2 functions? Maybe add another entry function on top of them?

I am fine with any refactoring that can make the logic obvious and straightforward. That will also be good for unit testing.